src/parser.rs

```diff
@@ -82,24 +82,30 @@ impl fmt::Display for ParserError {
 
 impl Error for ParserError {}
 
-/// SQL Parser
-pub struct Parser {
+//TODO: remove dead_code annotation when dialect will be in use
+#[allow(dead_code)]
+pub struct Parser<'a> {
```

Comment:
I'd rather use the dialect somewhere right away. This seems to be a good example:
```
        } else if self.parse_keyword(Keyword::AUTO_INCREMENT)
            && (self.dialect.is::<MySqlDialect>() || self.dialect.is::<GenericDialect>())
        {
            ColumnOption::DialectSpecific(vec![Token::make_keyword("AUTO_INCREMENT")])
        } else if self.parse_keyword(Keyword::AUTOINCREMENT)
            && (self.dialect.is::<SQLiteDialect>() || self.dialect.is::<GenericDialect>())
        {
            ColumnOption::DialectSpecific(vec![Token::make_keyword("AUTOINCREMENT")])
        } else {
```

Reply:
done